### PR TITLE
Bug Fixes and optimizations

### PR DESCRIPTION
**Pull request by Rboin**
Fixed a couple of things:

 - Moved WorldPanel logic to the SDLWorldPanel class.
 - Fixed the WorldPanel's mouse handler.
 - Optimized the loading of images to load directly onto a texture (Loading images onto a surface makes it so that we can't draw textures on top of one another. Fixed this)

### CODE REVIEW

File: src/ui/CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ set(SOURCE_FILES
         sdl/event/sdl_event_dispatcher.h
         sdl/event/sdl_mouse_event_dispatcher.h
         sdl/event/sdl_mouse_event_dispatcher.cpp
-        sdl/event/slot/sdl_mouse_event_slot.h sdl/event/slot/sdl_mouse_event_slot.cpp sdl/button/sdl_button.h sdl/button/sdl_button.cpp event/event_types.h sdl/panel/sdl_world_panel.cpp sdl/panel/sdl_world_panel.h sdl/event/sdl_event_types.h sdl/event/mouse_handler_world.h sdl/event/mouse_handler_world.cpp)
+        sdl/event/slot/sdl_mouse_event_slot.h sdl/event/slot/sdl_mouse_event_slot.cpp sdl/button/sdl_button.h sdl/button/sdl_button.cpp event/event_types.h sdl/panel/sdl_world_panel.cpp sdl/panel/sdl_world_panel.h sdl/event/sdl_event_types.h sdl/event/slot/mouse_handler_world.h sdl/event/slot/mouse_handler_world.cpp)
```
Comment (MarkVlamPersie) [inline src/ui/CMakeLists.txt]: Do files like this dont need to be on separate lines for readability?

**Comment by MarkVlamPersie** [comment]
One comment. Rest seems fine and it works on my machine.

**Comment by JeroenVinke** [comment]
http://programmingexcuses.com/ also lists "It works on my machine"

**Comment by MarkVlamPersie** [comment]
I have no comment to that.

**Pull request merged**